File: ios/Main/Factories/UseCaseFactory.swift
//
//  UseCaseFactory.swift
//  Uncompress
//
//  Created by Adriano Souza Costa on 17/03/21.
//

import Foundation

final class UseCaseFactory {
    
    private static let engines: [Extractable] = [
        ZipExtractor(),
        RarExtractor()
    ]
    
    static func extractUseCase() -> ExtractUseCase {
        return ExtractUseCase(engines: engines)
    }
    
    static func isProtectedUseCase() -> IsProctectedUseCase {
        return IsProctectedUseCase(engines: engines)
    }
    
}

File: android/src/main/java/com/uncompress/main/factories/UseCaseFactory.kt
package com.uncompress.main.factories

import com.uncompress.data.usecase.ExtractUseCase
import com.uncompress.data.usecase.IsProtectedUseCase
import com.uncompress.infra.adapters.rar.RarExtractor
import com.uncompress.infra.adapters.zip.ZipExtractor

class UseCaseFactory {

  companion object {
    private val engines = listOf(ZipExtractor(), RarExtractor())

    fun makeExtractUseCase(): ExtractUseCase {
      return ExtractUseCase(engines)
    }

    fun makeIsProtectedUseCase(): IsProtectedUseCase {
      return IsProtectedUseCase(engines)
    }
  }

}
